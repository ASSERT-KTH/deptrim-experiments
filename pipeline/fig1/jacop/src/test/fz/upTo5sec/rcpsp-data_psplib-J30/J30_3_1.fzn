predicate jacop_cumulative(array [int] of var int: s,array [int] of var int: d,array [int] of var int: r,var int: b);
array [1..2] of int: X_INTRODUCED_32 = [1,-1];
array [1..4] of int: X_INTRODUCED_90 = [7,8,9,6];
array [1..4] of int: X_INTRODUCED_91 = [6,9,9,10];
array [1..7] of int: X_INTRODUCED_93 = [8,10,5,1,5,3,8];
array [1..7] of int: X_INTRODUCED_94 = [1,6,10,10,6,8,8];
array [1..12] of int: X_INTRODUCED_96 = [10,7,5,1,1,7,9,5,3,5,3,2];
array [1..12] of int: X_INTRODUCED_97 = [4,2,10,10,6,10,8,1,10,5,1,9];
array [1..7] of int: X_INTRODUCED_99 = [2,2,7,4,10,9,7];
array [1..7] of int: X_INTRODUCED_100 = [7,7,1,7,9,10,5];
var 0..204: X_INTRODUCED_0;
var 0..204: X_INTRODUCED_1;
var 0..204: X_INTRODUCED_2;
var 0..204: X_INTRODUCED_3;
var 0..204: X_INTRODUCED_4;
var 0..204: X_INTRODUCED_5;
var 0..204: X_INTRODUCED_6;
var 0..204: X_INTRODUCED_7;
var 0..204: X_INTRODUCED_8;
var 0..204: X_INTRODUCED_9;
var 0..204: X_INTRODUCED_10;
var 0..204: X_INTRODUCED_11;
var 0..204: X_INTRODUCED_12;
var 0..204: X_INTRODUCED_13;
var 0..204: X_INTRODUCED_14;
var 0..204: X_INTRODUCED_15;
var 0..204: X_INTRODUCED_16;
var 0..204: X_INTRODUCED_17;
var 0..204: X_INTRODUCED_18;
var 0..204: X_INTRODUCED_19;
var 0..204: X_INTRODUCED_20;
var 0..204: X_INTRODUCED_21;
var 0..204: X_INTRODUCED_22;
var 0..204: X_INTRODUCED_23;
var 0..204: X_INTRODUCED_24;
var 0..204: X_INTRODUCED_25;
var 0..204: X_INTRODUCED_26;
var 0..204: X_INTRODUCED_27;
var 0..204: X_INTRODUCED_28;
var 0..204: X_INTRODUCED_29;
var 0..205: objective:: output_var;
var bool: X_INTRODUCED_59 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_60 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_61 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_62 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_63 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_64 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_65 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_66 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_67 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_68 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_69 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_70 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_71 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_72 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_73 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_74 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_75 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_76 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_77 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_78 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_79 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_80 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_81 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_82 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_83 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_84 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_86 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_87 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_88 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_89 ::var_is_introduced :: is_defined_var;
array [1..30] of var int: s:: output_array([1..30]) = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_8,X_INTRODUCED_9,X_INTRODUCED_10,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_13,X_INTRODUCED_14,X_INTRODUCED_15,X_INTRODUCED_16,X_INTRODUCED_17,X_INTRODUCED_18,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_21,X_INTRODUCED_22,X_INTRODUCED_23,X_INTRODUCED_24,X_INTRODUCED_25,X_INTRODUCED_26,X_INTRODUCED_27,X_INTRODUCED_28,X_INTRODUCED_29];
array [1..4] of var int: X_INTRODUCED_92 ::var_is_introduced  = [X_INTRODUCED_3,X_INTRODUCED_10,X_INTRODUCED_14,X_INTRODUCED_18];
array [1..7] of var int: X_INTRODUCED_95 ::var_is_introduced  = [X_INTRODUCED_1,X_INTRODUCED_9,X_INTRODUCED_17,X_INTRODUCED_21,X_INTRODUCED_23,X_INTRODUCED_24,X_INTRODUCED_25];
array [1..12] of var int: X_INTRODUCED_98 ::var_is_introduced  = [X_INTRODUCED_2,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_8,X_INTRODUCED_12,X_INTRODUCED_15,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_22,X_INTRODUCED_26,X_INTRODUCED_28,X_INTRODUCED_29];
array [1..7] of var int: X_INTRODUCED_101 ::var_is_introduced  = [X_INTRODUCED_0,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_11,X_INTRODUCED_13,X_INTRODUCED_16,X_INTRODUCED_27];
array [1..31] of var int: X_INTRODUCED_104 ::var_is_introduced  = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_8,X_INTRODUCED_9,X_INTRODUCED_10,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_13,X_INTRODUCED_14,X_INTRODUCED_15,X_INTRODUCED_16,X_INTRODUCED_17,X_INTRODUCED_18,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_21,X_INTRODUCED_22,X_INTRODUCED_23,X_INTRODUCED_24,X_INTRODUCED_25,X_INTRODUCED_26,X_INTRODUCED_27,X_INTRODUCED_28,X_INTRODUCED_29,objective];
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_0,X_INTRODUCED_4],-7);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_1,X_INTRODUCED_17],-1);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_2,X_INTRODUCED_3],-4);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_3,X_INTRODUCED_5],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_3,X_INTRODUCED_8],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_3,X_INTRODUCED_9],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_4,X_INTRODUCED_21],-2);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_5,X_INTRODUCED_6],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_5,X_INTRODUCED_7],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_5,X_INTRODUCED_10],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_6,X_INTRODUCED_11],-7);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_6,X_INTRODUCED_16],-7);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_7,X_INTRODUCED_12],-1);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_7,X_INTRODUCED_14],-1);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_8,X_INTRODUCED_11],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_8,X_INTRODUCED_24],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_9,X_INTRODUCED_15],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_9,X_INTRODUCED_27],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_10,X_INTRODUCED_18],-9);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_10,X_INTRODUCED_25],-9);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_11,X_INTRODUCED_13],-7);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_11,X_INTRODUCED_19],-7);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_12,X_INTRODUCED_13],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_12,X_INTRODUCED_16],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_13,X_INTRODUCED_23],-9);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_14,X_INTRODUCED_26],-9);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_14,X_INTRODUCED_29],-9);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_15,X_INTRODUCED_16],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_16,X_INTRODUCED_20],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_16,X_INTRODUCED_22],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_16,X_INTRODUCED_28],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_17,X_INTRODUCED_26],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_18,X_INTRODUCED_26],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_19,X_INTRODUCED_27],-8);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_20,X_INTRODUCED_23],-1);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_21,X_INTRODUCED_27],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_22,X_INTRODUCED_23],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_22,X_INTRODUCED_24],-10);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_23,X_INTRODUCED_25],-6);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_24,X_INTRODUCED_25],-8);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_25,X_INTRODUCED_29],-8);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_26,X_INTRODUCED_28],-5);
constraint array_bool_or([X_INTRODUCED_59,X_INTRODUCED_60],true);
constraint array_bool_or([X_INTRODUCED_61,X_INTRODUCED_62],true);
constraint array_bool_or([X_INTRODUCED_63,X_INTRODUCED_64],true);
constraint array_bool_or([X_INTRODUCED_65,X_INTRODUCED_66],true);
constraint array_bool_or([X_INTRODUCED_67,X_INTRODUCED_68],true);
constraint array_bool_or([X_INTRODUCED_69,X_INTRODUCED_70],true);
constraint array_bool_or([X_INTRODUCED_71,X_INTRODUCED_72],true);
constraint array_bool_or([X_INTRODUCED_73,X_INTRODUCED_74],true);
constraint array_bool_or([X_INTRODUCED_75,X_INTRODUCED_76],true);
constraint array_bool_or([X_INTRODUCED_77,X_INTRODUCED_78],true);
constraint array_bool_or([X_INTRODUCED_79,X_INTRODUCED_80],true);
constraint array_bool_or([X_INTRODUCED_81,X_INTRODUCED_82],true);
constraint array_bool_or([X_INTRODUCED_83,X_INTRODUCED_84],true);
constraint array_bool_or([X_INTRODUCED_86,X_INTRODUCED_87],true);
constraint array_bool_or([X_INTRODUCED_88,X_INTRODUCED_89],true);
constraint jacop_cumulative(X_INTRODUCED_92,X_INTRODUCED_91,X_INTRODUCED_90,15);
constraint jacop_cumulative(X_INTRODUCED_95,X_INTRODUCED_94,X_INTRODUCED_93,14);
constraint jacop_cumulative(X_INTRODUCED_98,X_INTRODUCED_97,X_INTRODUCED_96,15);
constraint jacop_cumulative(X_INTRODUCED_101,X_INTRODUCED_100,X_INTRODUCED_99,16);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_27,objective],-5);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_28,objective],-1);
constraint int_lin_le(X_INTRODUCED_32,[X_INTRODUCED_29,objective],-9);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_9,X_INTRODUCED_1],-6,X_INTRODUCED_59):: defines_var(X_INTRODUCED_59);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_1,X_INTRODUCED_9],-1,X_INTRODUCED_60):: defines_var(X_INTRODUCED_60);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_25,X_INTRODUCED_1],-8,X_INTRODUCED_61):: defines_var(X_INTRODUCED_61);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_1,X_INTRODUCED_25],-1,X_INTRODUCED_62):: defines_var(X_INTRODUCED_62);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_4,X_INTRODUCED_2],-2,X_INTRODUCED_63):: defines_var(X_INTRODUCED_63);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_2,X_INTRODUCED_4],-4,X_INTRODUCED_64):: defines_var(X_INTRODUCED_64);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_15,X_INTRODUCED_2],-10,X_INTRODUCED_65):: defines_var(X_INTRODUCED_65);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_2,X_INTRODUCED_15],-4,X_INTRODUCED_66):: defines_var(X_INTRODUCED_66);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_19,X_INTRODUCED_2],-8,X_INTRODUCED_67):: defines_var(X_INTRODUCED_67);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_2,X_INTRODUCED_19],-4,X_INTRODUCED_68):: defines_var(X_INTRODUCED_68);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_14,X_INTRODUCED_3],-9,X_INTRODUCED_69):: defines_var(X_INTRODUCED_69);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_3,X_INTRODUCED_14],-6,X_INTRODUCED_70):: defines_var(X_INTRODUCED_70);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_19,X_INTRODUCED_4],-8,X_INTRODUCED_71):: defines_var(X_INTRODUCED_71);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_4,X_INTRODUCED_19],-2,X_INTRODUCED_72):: defines_var(X_INTRODUCED_72);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_13,X_INTRODUCED_7],-9,X_INTRODUCED_73):: defines_var(X_INTRODUCED_73);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_7,X_INTRODUCED_13],-1,X_INTRODUCED_74):: defines_var(X_INTRODUCED_74);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_17,X_INTRODUCED_9],-10,X_INTRODUCED_75):: defines_var(X_INTRODUCED_75);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_9,X_INTRODUCED_17],-6,X_INTRODUCED_76):: defines_var(X_INTRODUCED_76);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_23,X_INTRODUCED_9],-6,X_INTRODUCED_77):: defines_var(X_INTRODUCED_77);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_9,X_INTRODUCED_23],-6,X_INTRODUCED_78):: defines_var(X_INTRODUCED_78);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_25,X_INTRODUCED_9],-8,X_INTRODUCED_79):: defines_var(X_INTRODUCED_79);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_9,X_INTRODUCED_25],-6,X_INTRODUCED_80):: defines_var(X_INTRODUCED_80);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_14,X_INTRODUCED_10],-9,X_INTRODUCED_81):: defines_var(X_INTRODUCED_81);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_10,X_INTRODUCED_14],-9,X_INTRODUCED_82):: defines_var(X_INTRODUCED_82);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_16,X_INTRODUCED_13],-10,X_INTRODUCED_83):: defines_var(X_INTRODUCED_83);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_13,X_INTRODUCED_16],-9,X_INTRODUCED_84):: defines_var(X_INTRODUCED_84);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_27,X_INTRODUCED_13],-5,X_INTRODUCED_86):: defines_var(X_INTRODUCED_86);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_13,X_INTRODUCED_27],-9,X_INTRODUCED_87):: defines_var(X_INTRODUCED_87);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_19,X_INTRODUCED_15],-8,X_INTRODUCED_88):: defines_var(X_INTRODUCED_88);
constraint int_lin_le_reif(X_INTRODUCED_32,[X_INTRODUCED_15,X_INTRODUCED_19],-10,X_INTRODUCED_89):: defines_var(X_INTRODUCED_89);
solve :: int_search(X_INTRODUCED_104,smallest,indomain_min,complete) minimize objective;
