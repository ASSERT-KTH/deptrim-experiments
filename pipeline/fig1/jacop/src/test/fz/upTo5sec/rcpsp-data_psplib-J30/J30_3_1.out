objective = 72;
s = array1d(1..30, [0, 0, 0, 4, 7, 10, 20, 20, 10, 11, 20, 27, 21, 37, 29, 17, 27, 1, 29, 34, 42, 9, 37, 47, 47, 55, 42, 46, 47, 63]);
----------
==========
